src/node/request_tracker.h

```diff
@@ -152,19 +162,22 @@ namespace aft
     kv::Consensus::SeqNo seqno_last_signature = -1;
     std::chrono::milliseconds time_last_signature =
       std::chrono::milliseconds(0);
+    mutable SpinLock lock;
 
     void insert(
       const crypto::Sha256Hash& hash,
       std::chrono::milliseconds time,
       std::multiset<Request*, RequestComp>& requests_,
       snmalloc::DLList<Request, std::nullptr_t, true>& requests_list_)
     {
-      CCF_ASSERT_FMT(
-        requests_list_.get_tail() == nullptr ||
-          requests_list_.get_tail()->time <= time,
-        "items not entered in the correct order. last:{}, time:{}",
-        requests_list_.get_tail()->time,
-        time);
+      if (
+        requests_list_.get_tail() != nullptr &&
+        requests_list_.get_tail()->time > time)
+      {
+        // Time is an not a precise measurement and can be different be
```

Comment:
```suggestion
        // Time is not a precise measurement and can be different
```

I don't follow what this time is for. If we're sometimes overriding it, why doesn't it always come from `requests_list_.get_tail()->time`? Separately, if I'm reading correctly this could be a `static` function since it doesn't interact with any member state?

Reply:
Added more info to the comment and made the function static